packages/coinstac-ui/app/render/components/pipelines/pipeline.jsx

```diff
@@ -194,30 +318,88 @@ class Pipeline extends Component {
             <ControlLabel>Name</ControlLabel>
             <FormControl
               type="input"
-              inputRef={(input) => { this.name = input; }}
+              onChange={(evt) => this.updatePipeline({ param: 'name', value: evt.target.value })}
+              defaultValue={ pipeline.name }
             />
           </FormGroup>
 
           <FormGroup controlId="description">
             <ControlLabel>Description</ControlLabel>
             <FormControl
               componentClass="textarea"
-              inputRef={(input) => { this.description = input; }}
+              onChange={(evt) => this.updatePipeline({ param: 'description', value: evt.target.value })}
+              defaultValue={ pipeline.description }
             />
           </FormGroup>
 
-          <Checkbox inline>Share this pipeline with other consortia</Checkbox>
+          <Row>
+            <Col sm={12}>
+              <ControlLabel>Owning Consortium</ControlLabel>
+            </Col>
+          </Row>
+
+          <Row>
+            <Col sm={12}>
+              {consortia &&
+              <SplitButton 
+                bsStyle="info" 
+                title={consortium ? consortium.name : "Consortia"} 
+                id="pipelineconsortia" 
+                onSelect={(value, e) => this.updatePipeline({ param: 'owningConsortium', value: value, consortiumName: e.target.innerHTML })}>
```

Comment:
I found a better way to write fewer anonymous functions like this by using a closure in the function instead. Change `updatePipeline` to the following:

```
updatePipeline(update) {
  return ({ target }) {
     if(update.param == 'owningConsortium')
      this.setState({ consortium: { id: target.checked, name: update.consortiumName }});
    this.setState(prevState => ({
      pipeline: { ...prevState.pipeline, [update.param]: target.checked },
    }));
  }
}
```

And call as follows: 
`onChange={this.updatePipeline({ param: 'shared' })}`

The concern is that anonymous functions in the `render()` method are recreated every time `render()` is called, creating more overhead. Using closures in this way get around that issue.

In fact, I'm looking at refactoring even further, starting with new component functions I write. It seems you can automatically bind functions to `this` by using arrow functions in their definitions and skip binding them in the constructor. See [here](https://github.com/facebook/react/issues/9851) and [here](https://medium.com/@machnicki/handle-events-in-react-with-arrow-functions-ede88184bbb). But I don't that's important now as all of the other components are written the way you have it currently.

Reply:
After looking at this more, I'd be fine if you didn't make this change. Seems needlessly complex. Though we probably should generally shoot for using closures over anonymous functions in general.